### PR TITLE
Clean up experimental dark tokens

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Part of https://github.com/Shopify/polaris-internal/issues/166
Part of https://github.com/Shopify/polaris-internal/issues/537

### WHAT is this pull request doing?

* remove all border gradient experimental tokens (we will be doing this internally for now since its not being used in other polaris components)
* remove `experimental` suffix from tooltip tokens
* add `bg-fill-secondary-selected` token
* `dark-experimental` theme changes

### How to 🎩
Demo of all button states 
<video src='https://github.com/user-attachments/assets/8fcb0c0f-2f47-4e12-8bb0-0a5d1a81d46f' />

Figma files with button states
* [base file](https://www.figma.com/design/pGwkMhydM8nURjBRHo9GVh/Dark-UI-Lab?m=auto&t=SeYk5v6iFy8cIDS8-1)
* [disabled states](https://www.figma.com/design/A8jMxmQVxJCvoeZ79Tx4c6/Dynamic-Top-bar-P.2?node-id=1802-25633&t=HH7bSvTidiRjNp7o-1)

Go through the flow, noticing all button states of:
1. Make an edit to product title
2. discard
3. Make an edit to product title
4. Save
5. Press other top bar buttons (user menu, notifications)

https://admin.web.web-zj1n.sophie-schneider.us.spin.dev/store/shop1/products/1

### CODE REVIEW

File: polaris-tokens/src/themes/dark.ts [hunk 1]
```diff
@@ -9,16 +9,18 @@ export const metaThemeDarkPartial = createMetaThemePartial({
...
     'color-text-secondary': {value: colors.gray[11]},
+    'color-text-disabled': {value: colors.gray[14]},
     'color-bg-surface-secondary-active': {value: colors.whiteAlpha[9]},
     'color-bg-surface-secondary-hover': {value: colors.whiteAlpha[6]},
     'color-bg-fill-transparent': {value: colors.whiteAlpha[8]},
     'color-bg-fill-brand': {value: colors.gray[1]},
     'color-text-brand-on-bg-fill': {value: colors.gray[15]},
     'color-bg-surface-hover': {value: colors.gray[14]},
-    'color-bg-fill-hover': {value: 'rgba(37, 37, 37, 1)'},
+    'color-bg-fill-hover': {value: colors.whiteAlpha[5]},
```
Comment (sophschneider) [inline polaris-tokens/src/themes/dark.ts:23]: @bernardojoaogarcia would you be able to double check these values? I think I cross referenced all of them but want a second pair of eyes

**Comment by alex-page** [comment]
Make sure this doesn't break the file editor 🙏

**Comment by sophschneider** [comment]
@alex-page fo sho! I already tophatted with the file editor next beta but might be worth @dustinmalik taking a look to double check

Screenshot 2024-07-29 at 10 20 10 PM

https://admin.web.web-zj1n.sophie-schneider.us.spin.dev/store/shop1/products/1

**Review by laurkim — APPROVED** [review]

**Review by sam-b-rose — APPROVED** [review]

**Review by bernardojoaogarcia — APPROVED** [review]
🙌 🙌

**Review by jtwee — APPROVED** [review]
🎩 of FileEditorModal LGTM

**Review by mateus — APPROVED** [review]

**Review by mrcthms — APPROVED** [review]

**Comment by alex-page** [comment]
Thanks @sophschneider was worried I had to do this lol

**Pull request merged**